### PR TITLE
[CI] enable nightly test runs

### PR DESCRIPTION
**Pull request by dpakach**
Fixes https://github.com/owncloud/ocis/issues/895

Enable nightly test runs in CI.

### CODE REVIEW

File: .drone.star [hunk 1]
```diff
@@ -129,7 +129,9 @@ def main(ctx):
     notify(ctx),
   ]
 
-  if '[docs-only]' in (ctx.build.title + ctx.build.message):
+  if context.build.cron != "":
```
Comment (phil-davis) [inline .drone.star:132]:
```suggestion
  if ctx.build.cron != "":
```

File: .drone.star [hunk 2]
```diff
@@ -129,7 +129,9 @@ def main(ctx):
     notify(ctx),
   ]
 
-  if '[docs-only]' in (ctx.build.title + ctx.build.message):
+  if ctx.build.cron != "":
+    pipelines = before
```
Comment (phil-davis) [inline .drone.star:133]: We could also do the `notify_pipelines` logic, like for the docs-only builds.
Then the drone crn will notify to rocket chat if something fails.

**Review by phil-davis — APPROVED** [review]
LGTM. We will only really know after merging.

@dpakach after I merge this, please

- do a PR with [docs-only] to verify that it works as expected.
- do a [tests-only] PR for something small, and we can verify that all pipelines run as expected in the CI of the PR
- then merge something and see what happens to both the merge CI and the cron CI

**Comment by wkloucek** [comment]
I'm having problems with this line: `if ctx.build.cron != "":` It prevents me to convert the starlark to yml. Is there a reason you didn't use if ctx.build.event == "cron":` ?

**Comment by phil-davis** [comment]
We found a post where the `build.cron` attribute was "documented". https://discourse.drone.io/t/starlark-context-not-have-some-detailed-info/6087/2

But now I see that the source code https://github.com/drone/drone-convert-starlark/blob/master/plugin/args.go does not mention "cron" in `func fromBuild`

Now I have to look for the things that `v.Event` can be.

I guess drone is a nerd-project - the source code is the documentation.

**Comment by dpakach** [comment]
@phil-davis seems like only these attributes are available when converting
```
"context"(build = "build"(branch = "master", commit = "", event = "push", message = "", ref = "refs/heads/master", source = "", target = ""), repo = "repo"(name = "", namespace = "", slug = ""))

```

**Comment by dpakach** [comment]
@wkloucek, I think in this case your approach works better, 
Please review this PR https://github.com/owncloud/ocis/pull/923

**Comment by wkloucek** [comment]
Yeah, finding reliable information on drone is really cumbersome. The main issue for me is, that I can not convert to yml anymore. 

Are you ok with using `if ctx.build.event == "cron":` as long as we don't specify cron names? It think this is what the `build.cron` should contain!?

We also could make a PR to @micbar because he is already "maintaining" a working drone cli https://github.com/micbar/drone-cli/commit/f0e9e08904a4ee2489a4f6cbb8d58af22f7ee813

**Comment by phil-davis** [comment]
After installing the latest drone-cli:
```
$ curl -L https://github.com/drone/drone-cli/releases/latest/download/drone_linux_amd64.tar.gz | tar zx
$ sudo install -t /usr/local/bin drone
$ drone --version
drone version 1.2.4
```

I get:
```
$ drone starlark
2020/11/23 15:21:10 starlark evaluation error:
Traceback (most recent call last):
  .drone.star:201:15: in main
Error: "build" struct has no .cron attribute
```
So at least the error message has more information.

I wonder how I install the latest `drone-convert-starlark` as a plugin to drone-cli?

**Comment by phil-davis** [comment]
https://github.com/drone/drone-cli/blob/master/drone/starlark/starlark.go

There is "starlark" knowledge built-in to the `drone-cli` repo. And that only has a subset of the `ctx` attributes that are available in the extension. Code-duplication - I guess the `drone-cli` stuff could have the other attributes added and maybe they will "just work".

**Comment by micbar** [comment]
see https://github.com/drone/drone-cli/pull/175

**Comment by phil-davis** [comment]
Also note for future readers.
https://github.com/owncloud/activity/blob/master/.drone.star#L969
```
								if (params['cron'] == ''):
									result['trigger']['ref'] = [
										'refs/pull/**',
										'refs/tags/**'
									]
									for branch in config['branches']:
										result['trigger']['ref'].append('refs/heads/%s' % branch)
								else:
									result['trigger']['cron'] = params['cron']

```

We have starlark code that allows pipelines to trigger on a specific "cron" value. At the moment the relevant repos just have a "nightly" cron job. And the starlark correctly generates pipelines that run only nightly - they do not run every PR.

So that stuff does work on the drone server, and we could potentially have some extended tests that run only for cron "weekly" etc.

That parses OK with local `drone starlark` and includes pipelines that have a section like:
```
trigger:
  cron:
  - nightly
```

So the local drone-cli know enough to allow that (or doesn't try hard enough to parse it)

**Pull request merged**